pkg/cloudprovider/providers/azure/azure_wrap.go

```diff
@@ -17,19 +17,23 @@ limitations under the License.
 package azure
 
 import (
+	"fmt"
 	"net/http"
-	"sync"
 	"time"
 
 	"github.com/Azure/azure-sdk-for-go/arm/compute"
 	"github.com/Azure/azure-sdk-for-go/arm/network"
 	"github.com/Azure/go-autorest/autorest"
-	"github.com/golang/glog"
 
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/kubernetes/pkg/cloudprovider"
 )
 
+var (
+	vmCacheTTL = 30 * time.Second
```

Comment:
VMs are pretty stable (except when we attach/detach disks) which is something the cloud provider does. So we can safety increase this to 2m or more.

Reply:
Should we consider VM being updated by the user? e.g. attaching another disk on Azure portal

Reply:
No - We have been very vocal that users shouldn't manually change the VM on azure portal/CLI. And VM cache expiration should take care of that.